src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs

```diff
@@ -272,6 +272,21 @@ public void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
             }
         }
 
+        public void ChangeDefaultCredentialsPolicy(SafeWinHttpHandle requestHandle, bool allowDefaultCredentials)
+        {
+            uint optionData = allowDefaultCredentials ?
+                Interop.WinHttp.WINHTTP_AUTOLOGON_SECURITY_LEVEL_LOW :
```

Comment:
Those guidelines in MSDN apply to direct use of WinHTTP and to prevent "credentials from being casually used to authenticate against an untrusted server.".

However, our object model for WinHttpHandler (and HttpClient/HttpClientHandler) is an opt-in model.  A developer is explicitly requesting that default credentials be used.  Further complicating this is the fact that `WINHTTP_AUTOLOGON_SECURITY_LEVEL_MEDIUM` only works when WinHTTP thinks the destination server is "Intranet".  And it figures that out by looking at the proxy bypass list that is normally passed into methods like WinHttpOpen, WInHttpSendRequest. However, our use of these WinHTTP functions is such that we always pass in NULL for those lists because we have a different object model for proxy interop.

In essence, using security level MEDIUM, ends up not working at all.  I.e., WinHTTP never thinks that the URI is "Intranet". So, setting security level to MEDIUM won't send default creds. I had a previous conversation with WinHTTP team about this and they recommended we use HIGH/LOW model.